src/main/java/mate/academy/intro/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package mate.academy.intro.service;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.intro.dto.cartitem.CreateCartItemRequestDto;
+import mate.academy.intro.dto.shoppingcart.ShoppingCartDto;
+import mate.academy.intro.exception.EntityNotFoundException;
+import mate.academy.intro.mapper.CartItemMapper;
+import mate.academy.intro.mapper.ShoppingCartMapper;
+import mate.academy.intro.model.Book;
+import mate.academy.intro.model.CartItem;
+import mate.academy.intro.model.ShoppingCart;
+import mate.academy.intro.model.User;
+import mate.academy.intro.repository.book.BookRepository;
+import mate.academy.intro.repository.cartitem.CartItemRepository;
+import mate.academy.intro.repository.shoppingcart.ShoppingCartRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserService userService;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public void addItem(CreateCartItemRequestDto createCartItemRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCartForUser();
+        Book book = bookRepository.findById(createCartItemRequestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException("Can't find book with this id:"
+                        + createCartItemRequestDto.getBookId()));
+        CartItem cartItem = cartItemMapper.toCartItem(createCartItemRequestDto);
+        cartItem.setBook(book);
+        cartItem.setShoppingCart(shoppingCart);
+        shoppingCart.getCartItems().add(cartItemRepository.save(cartItem));
+        shoppingCartRepository.save(shoppingCart);
+    }
```

Comment:
use cascade for saving cartItem
```suggestion
    @Override
    @Transactional
    public void addItem(CreateCartItemRequestDto createCartItemRequestDto) {
        ShoppingCart shoppingCart = getShoppingCartForUser();
        Book book = bookRepository.findById(createCartItemRequestDto.getBookId())
                .orElseThrow(() -> new EntityNotFoundException("Can't find book with this id:"
                        + createCartItemRequestDto.getBookId()));
        CartItem cartItem = cartItemMapper.toCartItem(createCartItemRequestDto);
        cartItem.setBook(book);
        cartItem.setShoppingCart(shoppingCart);
        shoppingCart.getCartItems().add(cartItem);
        shoppingCartRepository.save(shoppingCart);
    }
```